src/api/directives.md

```diff
@@ -62,7 +62,7 @@
 
   This directive triggers transitions when its condition changes.
 
-  When used together with `v-if`, `v-for` has a higher priority than v-if. See the [list rendering guide](../guide/list.html#v-for-with-v-if) for details.
+  When used together with `v-for`, `v-if` has a higher priority than `v-for`. We don't recommend using these two directives together on one element, see the [list rendering guide](../guide/list.html#v-for-with-v-if) for details.
```

Comment:
Half joking: Maybe we do this to save ourselves a bit just in case the next change reverses the behavior again?

```suggestion
  When used together, `v-if` has a higher priority than `v-for`. We don't recommend using these two directives together on one element — see the [list rendering guide](../guide/list.html#v-for-with-v-if) for details.
```

Reply:
I think this reads much better. The repetition of `v-for` didn't quite sound natural.